android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/fields/DDMFieldCheckboxMultipleView.java

```diff
@@ -0,0 +1,250 @@
+            }
+        }
+
+        if (this.field.isInline()) {
+            linearLayout.setOrientation(HORIZONTAL);
+        }
+
+        LayoutParams layoutParams = calculateLayoutParams(this.field.isInline(), this.field.showAsSwitcher());
+
+        List<Option> availableOptions = field.getAvailableOptions();
+
+        if (field.showAsSwitcher()) {
+            for (Option opt : availableOptions) {
+                Switch switchView = createSwitchView(opt, layoutParams);
+                addView(field, switchView);
+            }
+        } else {
+            for (Option opt : availableOptions) {
+                CheckBox checkBoxView = createCheckBoxView(opt, layoutParams);
+                addView(field, checkBoxView);
+            }
+        }
+
+        refresh();
+    }
+
+    private void addView(CheckboxMultipleField field, CompoundButton switchView) {
+        linearLayout.addView(switchView);
+        onChangedValueObservable = onChangedValueObservable.mergeWith(getMappedObservable(field, switchView));
+    }
+
+    private Observable<CheckboxMultipleField> getMappedObservable(final CheckboxMultipleField field,
+        final CompoundButton switchView) {
+
+        return RxCompoundButton.checkedChanges(switchView).doOnNext(new Action1<Boolean>() {
+            @Override
+            public void call(Boolean aBoolean) {
+                onCheckedChanged(switchView, aBoolean);
+            }
+        }).distinctUntilChanged().map(new Func1<Boolean, CheckboxMultipleField>() {
+            @Override
+            public CheckboxMultipleField call(Boolean aBoolean) {
+                return field;
+            }
+        });
+    }
+
+    private CheckBox createCheckBoxView(Option option, LayoutParams layoutParams) {
+
+        CheckBox checkBoxView = new CheckBox(getContext());
+        checkBoxView.setLayoutParams(layoutParams);
+        checkBoxView.setText(option.label);
+        checkBoxView.setTag(option);
+        checkBoxView.setOnCheckedChangeListener(this);
+        checkBoxView.setTypeface(getTypeface());
+        checkBoxView.setSaveEnabled(true);
+
+        if (this.field.isInline()) {
+            checkBoxView.setGravity(Gravity.TOP);
+        }
+
+        return checkBoxView;
+    }
+
+    private Switch createSwitchView(Option option, LayoutParams layoutParams) {
+
+        Switch switchView = new Switch(getContext());
+        switchView.setLayoutParams(layoutParams);
+        switchView.setText(option.label);
+        switchView.setTag(option);
+        switchView.setOnCheckedChangeListener(this);
+        switchView.setTypeface(getTypeface());
+        switchView.setSaveEnabled(true);
+
+        if (this.field.isInline()) {
+            switchView.setGravity(Gravity.TOP);
+        }
+
+        return switchView;
+    }
+
+    private LayoutParams calculateLayoutParams(Boolean isInline, Boolean showAsSwitcher) {
+
+        if (showAsSwitcher && !isInline) {
```

Comment:
extract local variable with the first layoutparam

Reply:
Yeah, better, I'll set just the changed props for the else case.